js/latoken.js

```diff
@@ -0,0 +1,715 @@
+        return this.milliseconds ();
+    }
+
+    async currentTime(){
+        const response = await this.publicGetExchangeInfoTime();
+        let time = this.safeString(response, 'time');
+        let timeSeconds = parseInt(response['unixTimeSeconds']);
+        let timeMiliseconds = parseInt(response['unixTimeMiliseconds']);
+        return {
+            'time': time,
+            'timeSeconds': timeSeconds,
+            'timeMiliseconds': timeMiliseconds,
+        }
+    }
+
+    async fetchMarkets (params = {}) {
+        let markets = await this.publicGetExchangeInfoPairs(params);
+        let result = [];
+            for (let i = 0; i < markets.length; i++) {
+                let market = markets[i];
+                let id = market['pairId'];
+                let baseId = market['baseCurrency'];
+                let quoteId = market['quotedCurrency'];
+                let base = this.commonCurrencyCode (baseId);
+                let quote = this.commonCurrencyCode (quoteId);
+                let symbol =  market['symbol'];
+                let precision = {
+                    'price': market['pricePrecision'],
+                    'amount': market['amountPrecision'],
+                };
+                let limits = {
+                    'amount': {
+                        'min': this.safeFloat (market, 'minQty'),
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': Math.pow (10, -precision['price']),
+                        'max': undefined,
+                    },
+                };
+                limits['cost'] = {
+                    'min': market['amountPrecision'],
+                    'max': undefined,
+                };
+                result.push ({
+                    'id': id,
+                    'symbol': symbol,
+                    'base': base,
+                    'quote': quote,
+                    'baseId': baseId,
+                    'quoteId': quoteId,
+                    'active': true,
+                    'precision': precision,
+                    'limits': limits,
+                    'info': market,
+                });
+            }
+        return result;
+    }
+
+    calculateFee (symbol, side, amount, price, takerOrMaker = 'taker') {
+        let market = this.markets[symbol];
+        let key = 'quote';
+        let rate = market[takerOrMaker];
+        let cost = amount * rate;
+        let precision = market['precision']['price'];
+        if (side === 'sell') {
+            cost *= price;
+        } else {
+            key = 'base';
+            precision = market['precision']['amount'];
+        }
+        cost = this.decimalToPrecision (cost, ROUND, precision, this.precisionMode);
+        return {
+            'type': takerOrMaker,
+            'currency': market[key],
+            'rate': rate,
+            'cost': parseFloat (cost),
+        };
+    }
+
+    async fetchBalance (params = undefined) {
```

Comment:
the signature of `fetchBalance` should be `fetchBalance (params = {})`

all signatures of the functions must match other implementations otherwise the transpiled php will complain.

Reply:
done